traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/tasks/AuthTokenFetchTask.java

```diff
@@ -11,14 +12,17 @@
 public class AuthTokenFetchTask extends Task {
 
     private TaskManager taskManager = new TaskManager();
-    private Context context;
-    private String apiKey;
     private Token token;
     private Error error;
 
-    public AuthTokenFetchTask(String apiKey, Context context) {
-        this.context = context;
-        this.apiKey = apiKey;
+    private NetworkTask.Route route;
+    private SharedPrefsWriteTask sharedPrefsWriteTask;
+
+    public AuthTokenFetchTask(Session session, Context context) {
+
+        //Writing encrypted token to shared prefs
+        sharedPrefsWriteTask= new SharedPrefsWriteTask(context, session.getApiKey());
```

Comment:
avoid instance variables holding onto tasks.

This creates memory leaks

Reply:
code formatting ^

Reply:
updated